File: C# .Net/JDI UI Framework/JDI/Commons/Pairs/Pair.cs
using System;
using System.Collections.Generic;

namespace Epam.JDI.Commons.Pairs
{
    public class Pair : Pair<string,string> { public Pair(string value1, string value2) : base(value1, value2) {} }

    public class Pair<TValue1, TValue2> : IEquatable<Pair<TValue1, TValue2>>
    {
        public bool Equals(Pair<TValue1, TValue2> other)
        {
            if (ReferenceEquals(null, other)) return false;
            if (ReferenceEquals(this, other)) return true;
            return EqualityComparer<TValue1>.Default.Equals(Value1, other.Value1) && EqualityComparer<TValue2>.Default.Equals(Value2, other.Value2);
        }

        public override int GetHashCode()
        {
            unchecked
            {
                return (EqualityComparer<TValue1>.Default.GetHashCode(Value1)*397) ^ EqualityComparer<TValue2>.Default.GetHashCode(Value2);
            }
        }

        public TValue1 Value1;
        public TValue2 Value2;

        public Pair(TValue1 value1, TValue2 value2)
        {
            Value1 = value1;
            Value2 = value2;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Pair<TValue1, TValue2>) obj);
        }
    }
}

File: Java/JDI/jdi-commons/src/main/java/com/epam/commons/pairs/Pair.java
package com.epam.commons.pairs;
/*
 * Copyright 2004-2016 EPAM Systems
 *
 * This file is part of JDI project.
 *
 * JDI is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * JDI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with JDI. If not, see <http://www.gnu.org/licenses/>.
 */

/**
 * Created by 12345 on 30.09.2014.
 */
public class Pair<TFirst, TSecond> {
    public TFirst key;
    public TSecond value;

    public Pair(TFirst value1, TSecond value2) {
        key = value1;
        value = value2;
    }

    public boolean equalTo(Pair<TFirst, TSecond> other) {
        return other != null && (this == other || key == other.key && value == other.value);
    }

    @Override
    public int hashCode() {
        return key.hashCode() ^ value.hashCode();
    }

    @Override
    public boolean equals(Object o) {
        if (o == null)
            return false;
        if (!(o instanceof PairString))
            return false;
        Pair<TFirst, TSecond> pairo = (Pair<TFirst, TSecond>) o;
        return this.key.equals(pairo.key)
                && this.value.equals(pairo.value);
    }

    @Override
    public String toString() {
        return key + ":" + value;
    }
}